server/game/core/chat/GameChat.ts

```diff
@@ -47,37 +47,72 @@ export class GameChat {
             return '';
         }
 
+        // split the format string by placeholders like {0}, {1}, etc.
         const fragments = format.split(/(\{\d+\})/);
-        return fragments.reduce((output, fragment) => {
+
+        const formatOutput = [];
+        for (const fragment of fragments) {
             const argMatch = fragment.match(/\{(\d+)\}/);
-            if (argMatch && args) {
-                const arg: MsgArg = args[argMatch[1]];
-                if (arg) {
-                    if (typeof arg === 'object' && 'message' in arg && arg.message) {
-                        return output.concat(arg.message);
-                    } else if (Array.isArray(arg)) {
-                        if (typeof arg[0] === 'string' && arg[0].includes('{')) {
-                            return output.concat(this.formatMessage(arg[0], arg.slice(1)));
-                        }
-                        return output.concat(this.formatArray(arg));
-                    } else if (typeof arg === 'object' && 'getShortSummary' in arg && arg.getShortSummary) {
-                        return output.concat(arg.getShortSummary());
-                    } else if (typeof arg === 'object' && 'format' in arg && 'args' in arg) {
-                        return output.concat(this.formatMessage(arg.format, arg.args));
-                    }
-                    return output.concat(arg);
+
+            let formattedFragment;
+            if (argMatch) {
+                formattedFragment = this.tryFormatPlaceholder(fragment, argMatch, args);
+            } else {
+                formattedFragment = fragment;
+            }
+
+            if (formattedFragment) {
+                if (Array.isArray(formattedFragment)) {
+                    formatOutput.push(...formattedFragment);
+                    continue;
+                }
+
+                const prevFragmentType = typeof formatOutput.at(-1);
+                const thisFragmentType = typeof formattedFragment;
+
+                if (prevFragmentType === 'string' && (thisFragmentType === 'string' || thisFragmentType === 'number')) {
+                    // concatenate simple strings and numbers together to reduce the number of elements sent over the wire
+                    formatOutput[formatOutput.length - 1] += formattedFragment;
+                } else if (thisFragmentType === 'number') {
+                    formatOutput.push(formattedFragment.toString());
+                } else {
+                    formatOutput.push(formattedFragment);
                 }
-            } else if (!argMatch && fragment) {
```

Comment:
We've lost this space separating that we used to do -- is there a reason we did this to begin with?